### DISCUSSION TITLE
Akka.NET 2024 Community Standups

### DISCUSSION THREAD

**Discussion by Aaronontheweb** [akkadotnet/akka.net#7041] [Community Standups]
Well well, we're onto our third year of continued Akka.NET Community Standups and this is our official discussion thread for posting updates about each of them. WE ARE ACTIVELY LOOKING FOR SPEAKERS, SO PLEASE LOOK AT #5690 IF YOU ARE INTERESTED IN CONTRIBUTING.

Akka.NET Community Standups are a once-a-month meeting that happens on the second Wednesday of each month and is live-streamed via [Petabridge’s YouTube channel](https://www.youtube.com/c/PetabridgeAcademy).

# Structure of Each Standup

1. **Recap of most recent releases (10 min)** - we should talk about the work that was performed since the last standup, its impact on end-users, and what we learned in the course of performing that work.
2. **Plans for future work (10 min)** - what are we working on next, why, and why should users care?
3. **Opportunities to contribute (5 min)** - what help would we like from the Akka.NET end-user community? 
4. **[Show and Tell](https://github.com/akkadotnet/akka.net/discussions/5690) (~20 min)** - give members of the Akka.NET team and community opportunities to showcase their work built on top of Akka.NET. See https://github.com/akkadotnet/akka.net/discussions/5690 for opportunities to apply.
5. **Q&A (~10 min)** - show and tell presenters along with the [Akka.NET](http://Akka.NET) team should take questions via YouTube Live from the audience.

**Standups are second Wednesday of each month starting at 12pm CST.**

**Comment by Aaronontheweb** [comment]
January 10th, 2024 meeting agenda is live here: https://petabridge.notion.site/1-10-2024-Akka-NET-Community-Standup-a5354ff7d76c4e73a4ed4c59e4d5eb61?pvs=4

We will be covering [Akka.Analyzers](https://github.com/akkadotnet/akka.analyzers) - our new set of Roslyn Analyzers being distributed as part of Akka.NET v1.5.15 and onward.

Watch live or recorded here: https://www.youtube.com/watch?v=1w828avJwPY

**Comment by shelbyfranklin** [comment]
May 15th, 2024 meeting agenda is live here: https://petabridge.notion.site/5-15-2024-Akka-NET-Community-Standup-0b6a6dd4ca2f4a0e974056afe744ebb5 

We'll be discussing recent Akka.NET releases, including a complete rewrite of Akka.Persistence.EventStore with the latest EventStore drivers and best practices, as well as major bug fixes and performance improvements. We're also introducing TurboMqtt - the fastest and most efficient Mqtt library in .NET. 

Watch live or recorded here: https://youtube.com/live/jOBFvXW1nUc

**Comment by qthanpt** [comment]
Are we need setup a MQTT Server before use TurboMQTT?

**Reply by Aaronontheweb** [reply]
You'll need to use an MQTT broker - before you can use TurboMQTT. Here's an example of how to set up [EMQX](https://www.emqx.io/) with Docker:

```powershell
docker run -d --name emqx-bench -p 1883:1883 -p 18083:18083 `
-e EMQX_MQTT__MAX_MQUEUE_LEN=100000 `
-e EMQX_FORCE_SHUTDOWN__ENABLE=false `
-e EMQX_MQTT__MAX_INFLIGHT=1000 `
emqx/emqx
```

**Reply by qthanpt** [reply]
You can describe different between TurboMQTT and other client MQTT liblary.

**Reply by Aaronontheweb** [reply]
TurboMqtt is significantly faster, lower in resource utilization, less error-prone, and works using backpressure-aware streams rather than event handlers, which makes it easier to reason about as a consumer. In the future we're also going to add support for QUIC, which is something that no other MQTT client does in .NET today.

**Comment by shelbyfranklin** [comment]
June 12th, 2024 meeting agenda is live here: https://petabridge.notion.site/6-12-2024-Akka-NET-Community-Standup-859bd45833a6470eaf9a7c8ef61741bf

We'll discuss the recent Akka.NET releases, highlighting three major updates since the last version. We'll also explore future plans for Akka.NET, with a focus on Ahead-Of-Time (AOT) Compilation for Akka.NET v1.6. Our goal is to enhance Akka.NET's efficiency and expand its compatibility across a wider range of platforms.

Watch live or recorded here: https://www.youtube.com/live/2QG5W0C8ujg

**Comment by shelbyfranklin** [comment]
July 17th, 2024 meeting agenda is live here: https://petabridge.notion.site/7-17-2024-Akka-NET-Community-Standup-fe9572aa75c04178a8b998df93d79888

In this video we'll discuss recent Akka.NET releases, highlighting MASSIVE bug fixes for LogFilter introduced in v1.5.21-v1.5.24, which broke Akka.NET logging almost entirely. We're also going to show the new test lab we’ve built in-house designed to help automate the reproduction of tricky distributed systems bugs and performance issues, a setup that helped us uncover several critical bugs in Akka.NET. 

Watch live or recorded here: https://youtube.com/live/y0zZEmWs8Jc

**Comment by shelbyfranklin** [comment]
It’s time to upgrade! Akka.NET v1.5.27 and related package updates resolve critical long-standing bugs. Plus, Akka.Persistence.Sql is now 350 times faster than its predecessor, making this a must-have upgrade. At our August 14th Community Standup, we'll discuss these recent Akka.NET releases and future plans, highlighting what these important updates mean for you.

Watch live or recorded here: https://www.youtube.com/watch?v=5Yop2sLQSVk

**Comment by shelbyfranklin** [comment]
Join us for this month’s Akka.NET Community Standup on September 11th, where we’ll discuss several key improvements and bug fixes introduced across multiple Akka.NET modules, including updates to Akka.Persistence.Sql, Akka.Hosting, and more. Plus, don’t miss our Show & Tell featuring Aaron Stannard as he demonstrates the prototype DotNetty replacement transport for Akka.Remote, designed to boost high-speed MQTT performance using Akka.Streams and System.Threading.Channels. If you’re working with IoT or MQTT-centric solutions, you’ll want to check this out!

Watch live or recorded here: https://youtube.com/live/Q6tHc7uDT3w

**Comment by shelbyfranklin** [comment]
Join us for this month's Akka.NET Community Standup on October 9th, where we'll discuss the key improvements and bug fixes in Akka.NET v1.5.30, including updates to Akka.Persistence.Sql, Akka.Persistence.Redis, and more!

Watch live or recorded here: https://youtube.com/live/fo8iYV6TusQ

**Comment by shelbyfranklin** [comment]
Join us for this month's Akka.NET Community Standup on November 13th at 12:00 PM CDT, where we'll cover key updates and significant fixes in Akka.NET v1.5.31, including the resolution of two 10-year-old critical bugs, plus improvements to Akka.Hosting, Akka.Management, and more!

Watch live or recorded here: https://youtube.com/live/SHOpu-tN6TI
